scripts/player_animation_controller.gd

```diff
@@ -0,0 +1,21 @@
+class_name PlayerAnimationController
+extends AnimationPlayer
+
+@export var animated_sprite: AnimatedSprite2D
+@export var player: CharacterBody2D
+
+func change_dir(dir: float):
+	if dir > 0:
+		animated_sprite.flip_h = false
+	elif dir < 0:
+		animated_sprite.flip_h = true
+
+func walking(dir: float):
+	change_dir(dir)
+	
+	if player.is_on_floor():
+		play(&"walking")
+
+func idle():
+	if player.is_on_floor():
+		play(&"idle")
```

Comment:
I dont think we shoudl be doing animations like this. I believe there is an animation tree that is a state machine

Reply:
Worth looking into for future projects, but is there any issue making the state machine in code for this one?